SendError and SendToError conflicts in the definition of `send` after #7417

zig version 6dfe9cc83ea75a663b4f6dfa488b95950207a1cf

```
date/master λ zig build
/gnu/store/a43966k9w3dxmxadwnijjxahwg1z4hm4-zig-master-0.7.1+6dfe9cc/lib/zig/std/os.zig:4926:18: error: expected type 'std.os.SendError!usize', found 'std.os.SendToError!usize'
    return sendto(sockfd, buf, flags, null, 0);
                 ^
/gnu/store/a43966k9w3dxmxadwnijjxahwg1z4hm4-zig-master-0.7.1+6dfe9cc/lib/zig/std/os.zig:4926:18: note: error set 'std.os.SendToError' cannot cast into error set 'std.os.SendError'
    return sendto(sockfd, buf, flags, null, 0);
                 ^
/gnu/store/a43966k9w3dxmxadwnijjxahwg1z4hm4-zig-master-0.7.1+6dfe9cc/lib/zig/std/os.zig:4781:5: note: 'error.AddressFamilyNotSupported' not a member of destination error set
    AddressFamilyNotSupported,
    ^
/gnu/store/a43966k9w3dxmxadwnijjxahwg1z4hm4-zig-master-0.7.1+6dfe9cc/lib/zig/std/os.zig:4784:5: note: 'error.SymLinkLoop' not a member of destination error set
    SymLinkLoop,
    ^
/gnu/store/a43966k9w3dxmxadwnijjxahwg1z4hm4-zig-master-0.7.1+6dfe9cc/lib/zig/std/os.zig:4787:5: note: 'error.NameTooLong' not a member of destination error set
    NameTooLong,
    ^
/gnu/store/a43966k9w3dxmxadwnijjxahwg1z4hm4-zig-master-0.7.1+6dfe9cc/lib/zig/std/os.zig:4790:5: note: 'error.FileNotFound' not a member of destination error set
    FileNotFound,
    ^
/gnu/store/a43966k9w3dxmxadwnijjxahwg1z4hm4-zig-master-0.7.1+6dfe9cc/lib/zig/std/os.zig:4791:5: note: 'error.NotDir' not a member of destination error set
    NotDir,
    ^
/gnu/store/a43966k9w3dxmxadwnijjxahwg1z4hm4-zig-master-0.7.1+6dfe9cc/lib/zig/std/os.zig:4794:5: note: 'error.NetworkUnreachable' not a member of destination error set
    NetworkUnreachable,
    ^
/gnu/store/a43966k9w3dxmxadwnijjxahwg1z4hm4-zig-master-0.7.1+6dfe9cc/lib/zig/std/os.zig:4800:5: note: 'error.SocketNotConnected' not a member of destination error set
    SocketNotConnected,
    ^
/gnu/store/a43966k9w3dxmxadwnijjxahwg1z4hm4-zig-master-0.7.1+6dfe9cc/lib/zig/std/os.zig:4802:5: note: 'error.AddressNotAvailable' not a member of destination error set
    AddressNotAvailable,
    ^
```

Also in the event loop:

```zig
lib/zig/std/event/loop.zig:1271:32: error: expected type 'std.os.SendError', found 'std.os.SendToError'
                else => return err,
```

--- a/lib/std/fs/file.zig
+++ b/lib/std/fs/file.zig
@@ -698,6 +698,8 @@ pub const File = struct {
             error.FastOpenAlreadyInProgress,
             error.MessageTooBig,
             error.FileDescriptorNotASocket,
+            error.NetworkUnreachable,
+            error.NetworkSubsystemFailed,
             => return self.writeFileAllUnseekable(in_file, args),
 
             else => |e| return e,

--- a/lib/std/os.zig
+++ b/lib/std/os.zig
@@ -4798,7 +4798,6 @@ pub const SendToError = SendError || error{
 
     /// The socket is not connected (connection-oriented sockets only).
     SocketNotConnected,
-    WouldBlock,
     AddressNotAvailable,
 };
 
@@ -4924,7 +4923,15 @@ pub fn send(
     flags: u32,
 ) SendError!usize {
     return sendto(sockfd, buf, flags, null, 0) catch |err| switch (err) {
-        error.AddressFamilyNotSupported, error.SymLinkLoop, error.NameTooLong, error.FileNotFound, error.NotDir, error.NetworkUnreachable, error.SystemResources, error.SocketNotConnected, error.WouldBlock, error.AddressNotAvailable => unreachable,
+        error.AddressFamilyNotSupported => unreachable,
+        error.SymLinkLoop => unreachable,
+        error.NameTooLong => unreachable,
+        error.FileNotFound => unreachable,
+        error.NotDir => unreachable,
+        error.NetworkUnreachable => unreachable,
+        error.SystemResources => unreachable,
+        error.SocketNotConnected => unreachable,
+        error.AddressNotAvailable => unreachable,
         else => return @errSetCast(SendError, err),
     };
 }

--- a/lib/std/os.zig
+++ b/lib/std/os.zig
@@ -4932,7 +4932,7 @@ pub fn send(
         error.NotDir => unreachable,
         error.NetworkUnreachable => unreachable,
         error.AddressNotAvailable => unreachable,
-        else => return @errSetCast(SendError, err),
+        else => |e| return e,
     };
 }
 

--- a/lib/std/os.zig
+++ b/lib/std/os.zig
@@ -4777,6 +4777,9 @@ pub const SendError = error{
 
     /// Network is unreachable.
     NetworkUnreachable,
+
+    /// The local network interface used to reach the destination is down.
+    NetworkSubsystemFailed,
 } || UnexpectedError;
 
 pub const SendToError = SendError || error{
@@ -4849,7 +4852,7 @@ pub fn sendto(
                     .WSAEHOSTUNREACH => return error.NetworkUnreachable,
                     // TODO: WSAEINPROGRESS, WSAEINTR
                     .WSAEINVAL => unreachable,
-                    .WSAENETDOWN => return error.NetworkUnreachable,
+                    .WSAENETDOWN => return error.NetworkSubsystemFailed,
                     .WSAENETRESET => return error.ConnectionResetByPeer,
                     .WSAENETUNREACH => return error.NetworkUnreachable,
                     .WSAENOTCONN => return error.SocketNotConnected,
@@ -4878,7 +4881,6 @@ pub fn sendto(
                 EMSGSIZE => return error.MessageTooBig,
                 ENOBUFS => return error.SystemResources,
                 ENOMEM => return error.SystemResources,
-                ENOTCONN => unreachable, // The socket is not connected, and no target has been given.
                 ENOTSOCK => unreachable, // The file descriptor sockfd does not refer to a socket.
                 EOPNOTSUPP => unreachable, // Some bit in the flags argument is inappropriate for the socket type.
                 EPIPE => return error.BrokenPipe,
@@ -4890,6 +4892,7 @@ pub fn sendto(
                 EHOSTUNREACH => return error.NetworkUnreachable,
                 ENETUNREACH => return error.NetworkUnreachable,
                 ENOTCONN => return error.SocketNotConnected,
+                ENETDOWN => return error.NetworkSubsystemFailed,
                 else => |err| return unexpectedErrno(err),
             }
         }

--- a/lib/std/os.zig
+++ b/lib/std/os.zig
@@ -4774,6 +4774,9 @@ pub const SendError = error{
     BrokenPipe,
 
     FileDescriptorNotASocket,
+
+    /// Network is unreachable.
+    NetworkUnreachable,
 } || UnexpectedError;
 
 pub const SendToError = SendError || error{
@@ -4789,9 +4792,6 @@ pub const SendToError = SendError || error{
     /// Returned when socket is AF_UNIX and the given path does not point to an existing file.
     FileNotFound,
     NotDir,
-
-    /// Network is unreachable.
-    NetworkUnreachable,
 
     /// The socket is not connected (connection-oriented sockets only).
     SocketNotConnected,
@@ -4888,6 +4888,7 @@ pub fn sendto(
                 ENOENT => return error.FileNotFound,
                 ENOTDIR => return error.NotDir,
                 EHOSTUNREACH => return error.NetworkUnreachable,
+                ENETUNREACH => return error.NetworkUnreachable,
                 ENOTCONN => return error.SocketNotConnected,
                 else => |err| return unexpectedErrno(err),
             }

--- a/lib/std/os.zig
+++ b/lib/std/os.zig
@@ -4923,7 +4923,10 @@ pub fn send(
     buf: []const u8,
     flags: u32,
 ) SendError!usize {
-    return sendto(sockfd, buf, flags, null, 0);
+    return sendto(sockfd, buf, flags, null, 0) catch |err| switch (err) {
+        error.AddressFamilyNotSupported, error.SymLinkLoop, error.NameTooLong, error.FileNotFound, error.NotDir, error.NetworkUnreachable, error.SystemResources, error.SocketNotConnected, error.WouldBlock, error.AddressNotAvailable => unreachable,
+        else => return @errSetCast(SendError, err),
+    };
 }
 
 pub const SendFileError = PReadError || WriteError || SendError;

--- a/lib/std/os.zig
+++ b/lib/std/os.zig
@@ -4891,6 +4891,7 @@ pub fn sendto(
                 ENOENT => return error.FileNotFound,
                 ENOTDIR => return error.NotDir,
                 EHOSTUNREACH => return error.NetworkUnreachable,
+                ENOTCONN => return error.SocketNotConnected,
                 else => |err| return unexpectedErrno(err),
             }
         }
@@ -4930,7 +4931,6 @@ pub fn send(
         error.NotDir => unreachable,
         error.NetworkUnreachable => unreachable,
         error.SystemResources => unreachable,
-        error.SocketNotConnected => unreachable,
         error.AddressNotAvailable => unreachable,
         else => return @errSetCast(SendError, err),
     };

--- a/lib/std/os.zig
+++ b/lib/std/os.zig
@@ -4793,9 +4793,6 @@ pub const SendToError = SendError || error{
     /// Network is unreachable.
     NetworkUnreachable,
 
-    /// Insufficient memory was available to fulfill the request.
-    SystemResources,
-
     /// The socket is not connected (connection-oriented sockets only).
     SocketNotConnected,
     AddressNotAvailable,
@@ -4930,7 +4927,6 @@ pub fn send(
         error.FileNotFound => unreachable,
         error.NotDir => unreachable,
         error.NetworkUnreachable => unreachable,
-        error.SystemResources => unreachable,
         error.AddressNotAvailable => unreachable,
         else => return @errSetCast(SendError, err),
     };

